pkgs/development/python-modules/coconut/default.nix

```diff
@@ -0,0 +1,45 @@
+{
+  lib,
+  buildPythonApplication,
+  fetchPypi,
+  fetchpatch,
+
+  dos2unix,
+  findutils,
+
+  cpyparsing,
+  ipykernel,
+  mypy,
+  pygments,
+  prompt_toolkit,
+  watchdog
+}:
+
+buildPythonApplication rec {
+  pname = "coconut";
+  version = "1.4.3";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "02rccpjnfxafp7csh49niba239yggp18cgjd2n9agalg71vfhlsh";
+  };
+
+  nativeBuildInputs = [ dos2unix findutils ];
+  propagatedBuildInputs = [ cpyparsing pygments prompt_toolkit ipykernel mypy watchdog ];
+
```

Comment:
There are no tests included. Please try to checkout from source and check if they have unit tests, and try to run them. Unit tests give a good indication that they package has a high degree of validity and correctness given the python package set.

If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors. Please see [pythonImportsCheck documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck) for more information.

```

----------------------------------------------------------------------
Ran 0 tests in 0.000s
```

Reply:
Tests added but not all are passing; looking into it and leaving this review comment open for now.